addon/components/table.ts

```diff
@@ -164,76 +161,36 @@ class TableComponent extends Component<TableArgs> {
         return !this.isLoading;
     }
 
-    /**
-     * CP that returns whether the table rows are clickable
-     *
-     * @type {boolean} clickableRows
-     * @memberof TableComponent
-     */
-    @or('onRowClick', 'onRowDoubleClick') clickableRows!: boolean;
+    get clickableRows(): boolean {
+        return !!this.args.onRowClick || !!this.args.onRowDoubleClick;
+    }
 
-    /**
-     * CP that returns whether a fully-loaded table is empty
-     *
-     * @type {boolean} isEmpty
-     * @memberof TableComponent
-     */
-    @and('noRows', 'notLoading') isEmpty!: boolean;
+    get isEmpty(): boolean {
+        return this.noRows && this.notLoading;
+    }
 
-    /**
-     * CP that returns whether there are non-rendered columns
-     *
-     * @type {boolean} hasHiddenColumns
-     * @memberof TableComponent
-     */
-    @lt('visibleColumns.length', 'args.columns.length') hasHiddenColumns!: boolean;
+    get hasHiddenColumns(): boolean {
+        return this.visibleColumns.length < this.args.columns.length;
+    }
 
-    /**
-     * CP that returns whether the table is currently panned
-     *
-     * @type {boolean} isColumnsPanned
-     * @memberof TableComponent
-     */
-    @gt('columnPanPosition', 0) isColumnsPanned!: boolean;
+    get isColumnsPanned(): boolean {
+        return this.columnPanPosition > 0;
+    }
 
-    /**
-     * CP that returns true when `containerWidth` >= `minFixedColTableWidth`
-     *
-     * @type {boolean} allowFixedCols
-     * @memberof TableComponent
-     */
-    @gte('containerWidth', 'minFixedColTableWidth') allowFixedCols!: boolean;
+    get allowFixedCols(): boolean {
+        return (this.containerWidth || 0) >= this.minFixedColTableWidth;
+    }
 
-    /**
-     * CP that returns which column should be used as the first
-     * visible column
-     *
-     * @type {ColumnValue} firstVisibleColumn
-     * @memberof TableComponent
-     */
-    @conditional('allowFixedCols', 'firstVisibleNonFixedColumn', 'visibleColumns.firstObject')
-    firstVisibleColumn!: ColumnValue;
+    get firstVisibleColumn() {
+        return this.allowFixedCols && this.firstVisibleNonFixedColumn
+            ? this.firstVisibleNonFixedColumn
+            : this.visibleColumns[0];
```

Comment:
this looks like its probably ok but i think the logic is slightly different from the original `conditional()` which was 
```ts
this.allowFixedCols ? this.firstVisibleNonFixedColumn : this.visibleColumns[0];
```
so i would just make sure everything still works as expected in this case